### PR TITLE
RawQuery: bugfixes, added short README

### PR DESCRIPTION
**Pull request by mobilunity-user**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,239 @@
...
+
+```ruby
+require 'ravendb'
+require 'models/product'
+```
+
+3. Configure document store:
+
+```ruby
+RavenDB.store.configure do |config|
```
Comment (ayende) [inline README.md:60]: How do I use multiple stores?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,239 @@
...
+  config.urls = ["database url"]
+  config.default_database = 'default database name'
```
Comment (ayende) [inline README.md:62]: Please name it `database`, not default database.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,239 @@
...
+session = RavenDB.store.open_session
+```
+
+4. Do some actions with documents:
+
+```ruby
+product = session.load('Products/1-A')
+product.in_stock = true
+
+session.store(product)
```
Comment (ayende) [inline README.md:78]: Why is this required if we loaded it from the server?

Comment (mobilunity-user) [reply README.md:78]: @ayende In NodeJS / Ruby / Python client we need to call store() for notify session that doc was changed

File: README.md (hunk 1 above)
Comment (mobilunity-user) [reply README.md:60]: @ayende 

store = RavenDB::DocumentStore.new(database urls, database)
store.configure

store.open_session do |session|
...
end

File: README.md (hunk 3 above)
Comment (ayende) [reply README.md:78]: Why? Aren't we doing change tracking there?

Comment (ayende) [reply README.md:78]: At least in Python client, we _are_ doing that, and this is what we want to have

Comment (mobilunity-user) [reply README.md:78]: @ayende we're doing change tracking but store also performs some additional checks + updates cached document info so anyway it should be called.
see how its done in Python client - i use it as my codebase

Comment (ayende) [reply README.md:78]: so why is the `store` needed?

Comment (mobilunity-user) [reply README.md:78]: - it builds default metadata for new documents (wasn't loaded before from this session)
- sets expected change vector(if was passed to store()) according to concurrency check mode
- checks if we already called store for this document
- checks if we already called delete for this document. documents marked as deleted can't be stored
- generates id (if document was new)
- updates current document state (raw entity + metadata + change vector)  in cache

Comment (ayende) [reply README.md:78]: * Why do we need to build metadata? We already loaded the document, and it had the metadata there.
* Setting the change vector is understood and how it works elsewhere.
* I'm talking specifically about the scenario in the doc here, where you are calling `store` after `load`.

Comment (mobilunity-user) [reply README.md:78]: In the scenario in the doc here, where we are calling store after load, the store() method do the following thinks:
- sets expected change vector (if was passed to store()) according to concurrency check mode
- checks if we already called store for this document
- checks if we already called delete for this document. documents marked as deleted can't be stored
- updates current document state (raw entity + metadata + change vector) in cache

Comment (ayende) [reply README.md:78]: Is this required? What happens if we'll drop it, will everything still work?

Comment (mobilunity-user) [reply README.md:78]: @ayende I think better to ask this question to Idan as he's an author of Python client. I just took this logic and ported it to nodeJS. After i did it, there were just small changes (related to change vectors handling)

Comment (mobilunity-user) [reply README.md:78]: @ayende From my point of view it's just a part of Raven concept which prefers short sessions with less amount of requests. So send delete & update or double/triple updates of the same document are forcibly denied

Comment (ayende) [reply README.md:78]: I spoke to @IdanHaim about this, according to him, this is not required.

Comment (ayende) [reply README.md:78]: And yes, short sessions are preferred, but the process is usually:
`load`, modify, `save_changes`
